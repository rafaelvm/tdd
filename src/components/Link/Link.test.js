import React from "react";
import renderer from "react-test-renderer";
import Link from "./Link";

test("Mudar a classe do componente quando tiver com hover", () => {
  const { act, create } = renderer;

  const component = create(<Link page="http://facebook.com">Facebook</Link>);

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  act(() => {
    tree.props.onMouseEnter();
  });

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  act(() => {
    tree.props.onMouseLeave();
  });

  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
