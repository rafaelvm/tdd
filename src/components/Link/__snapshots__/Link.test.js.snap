// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mudar a classe do componente quando tiver com hover 1`] = `
<a
  className="normal"
  href={0}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`Mudar a classe do componente quando tiver com hover 2`] = `
<a
  className="hovered"
  href={0}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;

exports[`Mudar a classe do componente quando tiver com hover 3`] = `
<a
  className="normal"
  href={0}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Facebook
</a>
`;
